build/configs/imxrt1020-evk/Make.defs

```diff
@@ -130,19 +138,35 @@ ifeq "$(ARCHMINOR)" "9"
 endif
 endif
 
-endif
 ARCHDEFINES =
 ifeq ($(CONFIG_ARM_UNWIND),y)
   ARCHDEFINES = -fasynchronous-unwind-tables -mno-sched-prolog
 endif
 ARCHPICFLAGS = -fpic -msingle-pic-base -mpic-register=r10
 
-CFLAGS = $(ARCHCFLAGS) $(ARCHWARNINGS) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe -ffunction-sections -fdata-sections
+CFLAGS = $(ARCHCFLAGS) $(ARCHWARNINGS) $(ARCHOPTIMIZATION) $(ARCHINCLUDES) $(EXTRADEFINES) -pipe 
+CFLAGS += -ffunction-sections -fdata-sections $(ARCHDEFINES)
+CFLAGS += $(ARCHCPUFLAGS)
 CPICFLAGS = $(ARCHPICFLAGS) $(CFLAGS)
-CXXFLAGS = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHXXINCLUDES)
-CXXFLAGS += $(ARCHDEFINES) $(EXTRADEFINES) -pipe -std=c++11 -DCLOCK_MONOTONIC -D__TINYARA__
-CXXFLAGS += -fexceptions -fcheck-new -frtti
-CXXFLAGS += -pedantic -D_DEBUG -D_LIBCPP_BUILD_STATIC -ffunction-sections -fdata-sections -D__GLIBCXX__
+CXXFLAGS = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHXXINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe
+
+ifneq ($(CONFIG_APP_BINARY_SEPARATION),y)
```

Comment:
@abhishek-samsung modified cxx support on app separation so that I think this should be CONFIG_LIBXX, not CONFIG_APP_BINARY_SEPARATION.